Steps4Impact/Journey/Cells/JourneyDetailCell.swift

```diff
@@ -0,0 +1,156 @@
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+ * OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **/
+
+import UIKit
+
+protocol JourneyDetailDelegate: class {
+  func didGetShareText(_ shareText: String)
+}
+
+struct JourneyDetailContext: CellContext {
+  var identifier: String = JourneyDetailCell.identifier
+  var milestone: Milestone?
+}
+
+class JourneyDetailCell: ConfigurableTableViewCell {
+  static var identifier = "JourneyDetailCell"
+  weak var delegate: JourneyDetailDelegate?
+  
+  private let titleLabel = UILabel(typography: .headerTitle)
+  private let subtitleLabel = UILabel(typography: .title)
+  private let bodyText: UITextView = {
+    let textView = UITextView()
+    textView.isUserInteractionEnabled = true
+    textView.isEditable = false
+    textView.isScrollEnabled = false
+    textView.contentInset = .zero
+    return textView
+  }()
+  
+  override func commonInit() {
+    super.commonInit()
+    
+    contentView.addSubview(titleLabel) {
+      $0.top.equalToSuperview().offset(Style.Padding.p20)
+      $0.leading.equalToSuperview().offset(Style.Padding.p24)
+      $0.trailing.equalToSuperview().inset(Style.Padding.p24)
+    }
+    
+    contentView.addSubview(subtitleLabel) {
+      $0.top.equalTo(titleLabel.snp.bottom).offset(Style.Padding.p8)
+      $0.leading.equalToSuperview().offset(Style.Padding.p24)
+      $0.trailing.equalToSuperview().inset(Style.Padding.p24)
+    }
+    
+    contentView.addSubview(bodyText) {
+      $0.top.equalTo(subtitleLabel.snp.bottom).offset(Style.Padding.p16)
+      $0.leading.equalToSuperview().offset(Style.Padding.p24)
+      $0.trailing.bottom.equalToSuperview().inset(Style.Padding.p24)
+      $0.bottom.equalToSuperview()
+    }
+    
+    titleLabel.textColor = Style.Colors.FoundationGrey
+    subtitleLabel.textColor = Style.Colors.FoundationGrey
+  }
+  
+  func configure(context: CellContext) {
+    guard let detailContext = context as? JourneyDetailContext,
+      let milestone = detailContext.milestone else { return }
+    
+    titleLabel.text = milestone.name
+    subtitleLabel.text = milestone.subtitle
+    bodyText.attributedText = getContentAttributedText(from: milestone)
+  }
+}
+
+extension JourneyDetailCell {
+  func getContentAttributedText(from milestone: Milestone) -> NSMutableAttributedString {
+    let htmlstring = milestone.content
+    let stringtext = htmlstring.html2String
+    let result = NSMutableAttributedString()
+    let split = stringtext.components(separatedBy: "https")
+    
+    if split.count > 1 {
+      let stringURL = "https\(String(split[1]))"
```

Comment:
since its wrapped in the if statement its fine - but we also have a collections extension for safe index access: https://github.com/AgaKhanFoundation/WCF-iOS/blob/master/Steps4Impact/Support/Collection.swift

You could do `split[safe: 1]` and it returns an optional and won't crash when index out of bounds

Reply:
@samisuteria Is that because it is not thread safe and could change after checking line 99?

Reply:
No its more of a style thing. I prefer the "assert and guard unwrap optionals at top" style instead of wrapping inside "if/elses" branches inside the function

Reply:
With the `split[safe: 1]` method - you don't need to have the `if` statement but you do need to unwrap the optional

Reply:
Fixed!